tools/libgcopdd.py

```diff
@@ -0,0 +1,104 @@
+import sys
+import os
+from os import path
+import gzip
+
+
+class Event:
+	def __init__(self, event_id=None, **loaded_entries):
+		self.event_id = event_id
+		self.loaded_entries = loaded_entries
+	
+	def get(self, entry_type, load):
+		result = self.loaded_entries.get(entry_type)
+		if result is None:
+			self.loaded_entries[entry_type] = result = load(self.event_id)
+		return result
+
+
+class UserError(Exception):
+	def __init__(self, msg):
+		self.msg = msg
+
+
+def args_to_events(args):
+	def inner():
+		for arg in args:
+			if path.isdir(arg):
+				# this is a dump
+				for basename in os.listdir(arg):
+					event_id, entry_type, *_ = basename.split('.')
+					if entry_type == 'request':
+						yield arg + '/' + event_id
+			elif path.exists(arg):
+				# this is an event list
```

Comment:
It would be better to explain things -> "this is an event list stored in a file, so we will load its content"